src/modules/display/hourly.rs

```diff
@@ -219,42 +218,44 @@ impl HourlyForecast {
 					.map(|x| x.ceil() as u8)
 					.collect::<Vec<u8>>(),
 			};
-		} else if current_hour != day_end_idx - 1 || day_index == 6 {
-			temperatures = &weather.hourly.temperature_2m[day_start_idx..=day_end_idx];
-			weather_codes = &weather.hourly.weathercode[day_start_idx..=day_end_idx];
+		// If it's the end of one day we show the weather of the next day
+		} else if current_hour == day_end_idx - 1 {
+			temperatures = &weather.hourly.temperature_2m[next_day_start_idx..=next_day_end_idx];
+			weather_codes = &weather.hourly.weathercode[next_day_start_idx..=next_day_end_idx];
 			precipitation = match params.config.units.precipitation {
 				Precipitation::probability => {
-					weather.hourly.precipitation_probability[day_start_idx..=day_end_idx].to_vec()
+					weather.hourly.precipitation_probability[next_day_start_idx..=next_day_end_idx].to_vec()
 				}
-				_ => weather.hourly.precipitation[day_start_idx..=day_end_idx]
+				_ => weather.hourly.precipitation[next_day_start_idx..=next_day_end_idx]
 					.iter()
 					.map(|x| x.ceil() as u8)
 					.collect::<Vec<u8>>(),
 			};
 		} else {
-			temperatures = &weather.hourly.temperature_2m[next_day_start_idx..=next_day_end_idx];
-			weather_codes = &weather.hourly.weathercode[next_day_start_idx..=next_day_end_idx];
+			temperatures = &weather.hourly.temperature_2m[day_start_idx..=day_end_idx];
+			weather_codes = &weather.hourly.weathercode[day_start_idx..=day_end_idx];
 			precipitation = match params.config.units.precipitation {
 				Precipitation::probability => {
-					weather.hourly.precipitation_probability[next_day_start_idx..=next_day_end_idx].to_vec()
+					weather.hourly.precipitation_probability[day_start_idx..=day_end_idx].to_vec()
 				}
-				_ => weather.hourly.precipitation[next_day_start_idx..=next_day_end_idx]
+				_ => weather.hourly.precipitation[day_start_idx..=day_end_idx]
 					.iter()
 					.map(|x| x.ceil() as u8)
 					.collect::<Vec<u8>>(),
 			};
 		};
 
-		let time_indicator_col = match day_index {
-			0 => Some(
+		let time_indicator_col = if day_index == 0 && params.config.gui.graph.time_indicator {
```

Comment:
It's mainly this line.

Rest are incremental refactors of the related block:
- the condition that determines if it's the hour of a day is flipped
to opt-into the last hour, which should be somewhat easier to follow.
- variable declarations are sorted.
- comments are updated.